rcl/src/rcl/logging.c

```diff
@@ -64,10 +67,13 @@ rcl_logging_ext_lib_output_handler(
   const char * format, va_list * args);
 
 rcl_ret_t
-rcl_logging_configure(const rcl_arguments_t * global_args, const rcl_allocator_t * allocator)
+rcl_logging_init(const rcl_arguments_t * global_args, const rcl_allocator_t * allocator)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(global_args, RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(allocator, RCL_RET_INVALID_ARGUMENT);
+  if (rcutils_atomic_fetch_add_uint64_t(&__rcl_logging_init_count, 1)) {
```

Comment:
Why are we counting and not just using a flag? This function is marked as not thread-safe, and `rcl_logging_fini` as well. So we don't have to handle the case where this is called multiple times concurrently. It's up to the code above this to ensure it isn't called in a thread unsafe way.

Reply:
Ok, I see why a counter, but why atomic?